frontend/catalyst/compiler.py

```diff
@@ -106,11 +108,26 @@ def __deepcopy__(self, memo):
 
     def get_pipelines(self) -> List[Tuple[str, List[str]]]:
         """Get effective pipelines"""
-        if self.pipelines:
-            return self.pipelines
-        elif self.async_qnodes:
-            return DEFAULT_ASYNC_PIPELINES  # pragma: nocover
-        return DEFAULT_PIPELINES
+        if not self.quantum_circuit_transforms:
+            if self.pipelines:
+                return self.pipelines
+            elif self.async_qnodes:
+                return DEFAULT_ASYNC_PIPELINES  # pragma: nocover
+            return DEFAULT_PIPELINES
+
+        else:  # if self.quantum_circuit_transforms:
+            # The quantum circuit transform pipeline (performing the peephole optimizations,
+            # e.g. cancel neighbouring Hadamards) is placed after QuantumCompilation and before
+            # Bufferization.
+            _DEFAULT_PIPELINES = DEFAULT_PIPELINES.copy()
+            _DEFAULT_PIPELINES.insert(2, QUANTUM_CIRCUIT_TRANSFORMS)
+            _DEFAULT_ASYNC_PIPELINES = DEFAULT_ASYNC_PIPELINES.copy()
+            _DEFAULT_ASYNC_PIPELINES.insert(2, QUANTUM_CIRCUIT_TRANSFORMS)
```

Comment:
This seems like a quick workaround than a proper fix :) Why does it need to be placed after QC and before Bufferization? How can this info be encoded (may be in`active_passes`) and utilized here to remove this patch?

Reply:
I agree, the decision on where to place the peephole pass is completely arbitrary at the moment, and maybe could use some discussion. 

This is also upon second thought not a good way to implement this: the handling of which pipelines to run should be outside this function, likely somewhere in the peripherals of `active_passes`.

Reply:
I just realized this workaround isn't actually needed, since the `QUANTUM_CIRCUIT_TRANSFORMS` pipeline is initialized to empty list anyway. I have removed it.